internal/converter/internal/staticconvert/internal/build/converter_discoveryprocessor.go

```diff
@@ -0,0 +1,105 @@
+package build
+
+import (
+	"fmt"
+
+	"github.com/grafana/agent/internal/component/discovery"
+	"github.com/grafana/agent/internal/component/otelcol"
+	otelcol_discovery "github.com/grafana/agent/internal/component/otelcol/processor/discovery"
+	"github.com/grafana/agent/internal/converter/diag"
+	"github.com/grafana/agent/internal/converter/internal/common"
+	"github.com/grafana/agent/internal/converter/internal/otelcolconvert"
+	"github.com/grafana/agent/internal/converter/internal/prometheusconvert"
+	"github.com/grafana/agent/internal/converter/internal/prometheusconvert/build"
+	prometheus_component "github.com/grafana/agent/internal/converter/internal/prometheusconvert/component"
+	"github.com/grafana/agent/internal/static/traces/promsdprocessor"
+	"github.com/grafana/river/scanner"
+	prom_config "github.com/prometheus/prometheus/config"
+	"go.opentelemetry.io/collector/component"
+	"gopkg.in/yaml.v3"
+)
+
+func init() {
+	// Do not append this to the converter because it is a custom processor
+	// from static mode and not a native otel processor.
+}
```

Comment:
open question: the init function couldn't register the converter anymore anyway since it's in a different package. You can _probably_ remove this function but it might be a good idea to keep just to be explicit.